LadybugTools_oM/AnalysisPeriod.cs

```diff
@@ -0,0 +1,58 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+
+using BH.oM.Base;
+using System.ComponentModel;
+
+namespace BH.oM.LadybugTools
+{
+    public class AnalysisPeriod : ILadybugTools
+    {
+        [Description("The Ladybug datatype of this object, used for deserialisation.")]
+        public virtual string Type { get; set; } = "AnalysisPeriod";
+        
+        [Description("The start month.")]
+        public virtual int StMonth { get; set; } = 1;
```

Comment:
Please put variable names in full - `StartMonth` etc.

Reply:
This is mirroring the naming convention for Ladybug objects. Putting this stuff into the Adapter and hiding the name converts would enable this to be named per expected BHoM conventions, but for now the auto-convert by changing string case works so I'm requesting a bit of dispensation on this one for now. Probably applicable to a lot of other object properties in this PR too.